siriuspy/siriuspy/clientarch/pvarch.py

```diff
@@ -113,55 +114,155 @@ def __str__(self):
 class PVData:
     """Archive PV Data."""
 
+    PARALLEL_QUERY_BIN_INTERVAL = 12*60*60  # 12h
+
     def __init__(self, pvname, connector=None):
         """."""
-        self.pvname = pvname
-        self.connector = connector
-        self.timestamp_start = None
-        self.timestamp_stop = None
-        self.timestamp = None
-        self.value = None
-        self.status = None
-        self.severity = None
+        self._pvname = pvname
+        self._connector = connector
+        self._time_start = None
+        self._time_stop = None
+        self._timestamp = None
+        self._value = None
+        self._status = None
+        self._severity = None
 
     @property
-    def connected(self):
-        """."""
-        return self.connector and self.connector.connected
+    def pvname(self):
+        """PVName."""
+        return self._pvname
 
     @property
     def request_url(self):
-        """."""
+        """Request url."""
         self.connect()
         url = self.connector.getData(
-            self.pvname, self.timestamp_start,
-            self.timestamp_stop, get_request_url=True)
+            self.pvname,
+            self._time_start.get_iso8601(),
+            self._time_stop.get_iso8601(),
+            get_request_url=True)
         return url
 
     @property
     def is_archived(self):
-        """."""
+        """Is archived."""
         self.connect()
         req = self.connector.getPVDetails(self.pvname)
         if not req.ok:
             return False
         return True
 
     def connect(self):
-        """."""
+        """Connect."""
         if self.connector is None:
-            self.connector = _ClientArchiver()
+            self._connector = _ClientArchiver()
+
+    @property
+    def connector(self):
+        """Connector."""
+        return self._connector
+
+    @property
+    def connected(self):
+        """Check connected."""
+        return self.connector and self.connector.connected
+
+    @property
+    def timestamp_start(self):
+        """Timestamp start."""
+        if not self._time_start:
+            return None
+        return self._time_start.timestamp()
+
+    @timestamp_start.setter
+    def timestamp_start(self, new_timestamp):
+        if not isinstance(new_timestamp, (float, int)):
+            raise TypeError('expected argument of type float or int')
+        self._time_start = _Time(timestamp=new_timestamp)
+
+    @property
+    def time_start(self):
+        """Time start."""
+        return self._time_start
+
+    @time_start.setter
+    def time_start(self, new_time):
+        if not isinstance(new_time, _Time):
+            raise TypeError('expected argument of type Time')
+        self._time_start = new_time
+
+    @property
+    def timestamp_stop(self):
+        """Timestamp stop."""
+        if not self._time_stop:
+            return None
+        return self._time_stop.timestamp()
+
+    @timestamp_stop.setter
+    def timestamp_stop(self, new_timestamp):
```

Comment:
Same here, I think `timestamp_(start|stop)` should be strings. You could accept float or integers in `time_(start|stop)`.

Reply:
idem of previous comment.